symplyphysics/laws/relativistic/proper_time_for_timelike_intervals.py

```diff
@@ -33,11 +33,7 @@
 def calculate_proper_time(
     spacetime_interval_: Quantity,
 ) -> Quantity:
-    _, imag = spacetime_interval_.scale_factor.as_real_imag()
-    try:
-        assert_equal(imag, 0)
-    except AssertionError as e:
-        raise ValueError("Interval must be real") from e
+    assert_equal(im(spacetime_interval_.scale_factor), 0)
```

Comment:
Don't use `assert_equal`. It checks with tolerance and here it should be strictly zero to pass as real number.